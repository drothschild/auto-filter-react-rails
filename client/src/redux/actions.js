export const CHANGE_SEARCH_TERM = 'CHANGE_SEARCH_TERM';
export const REQUEST_PEOPLE = 'REQUEST_PEOPLE';
export const RECEIVE_PEOPLE = 'RECEIVE_PEOPLE';
