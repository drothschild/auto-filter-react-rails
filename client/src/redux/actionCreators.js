import axios from 'axios';
import { CHANGE_SEARCH_TERM, RECEIVE_PEOPLE, REQUEST_PEOPLE } from './actions';

export const changeSearchTerm = term => ({
  type: CHANGE_SEARCH_TERM,
  searchTerm: term,
});

export const receivePeople = data => ({
  type: RECEIVE_PEOPLE,
  data,
});

const requestPeople = () => ({
  type: REQUEST_PEOPLE,
});

export const fetchPeople = () => (dispatch) => {
  dispatch(requestPeople);
  return axios
    .get('/api/v1/people')
    .then((res) => {
      dispatch(receivePeople(res.data));
    })
    .catch((error) => {
      console.error('axios error', error); // eslint-disable-line no-console
    });
};
