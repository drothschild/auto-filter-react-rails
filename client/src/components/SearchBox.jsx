import React from 'react';
import { connect } from 'react-redux';
import { changeSearchTerm } from '../redux/actionCreators';

const SearchBox = ({ searchTerm, handleSearchTermChange }) => (
  <div className="search-box">
    <input type="text" placeholder="Search" value={searchTerm} onChange={handleSearchTermChange} />
  </div>
);

SearchBox.defaultProps = {
  searchTerm: '',
  handleSearchTermChange: () => '',
};

const mapStateToProps = state => ({ searchTerm: state.searchTerm });

const mapDispatchToProps = dispatch => ({
  handleSearchTermChange(event) {
    dispatch(changeSearchTerm(event.target.value));
  },
});

export const Unwrapped = SearchBox;
export default connect(mapStateToProps, mapDispatchToProps)(SearchBox);
