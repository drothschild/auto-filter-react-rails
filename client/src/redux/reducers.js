import { combineReducers } from 'redux';
import { REQUEST_PEOPLE, RECEIVE_PEOPLE, RECEIVE_ERRORS, CHANGE_SEARCH_TERM } from './actions';

const people = (state = { data: [], isFetching: true }, action) => {
  switch (action.type) {
    case REQUEST_PEOPLE:
      return { data: [], isFetching: true };
    case RECEIVE_PEOPLE:
      return { data: action.data, isFetching: false };
    default:
      return state;
  }
};

const errors = (state = [], action) => {
  switch (action.type) {
    case RECEIVE_ERRORS:
      return action.errors;
    default:
      return [];
  }
};

const searchTerm = (state = '', action) => {
  switch (action.type) {
    case CHANGE_SEARCH_TERM:
      return action.searchTerm;
    default:
      return state;
  }
};

const rootReducer = combineReducers({
  errors,
  people,
  searchTerm,
});

export default rootReducer;
