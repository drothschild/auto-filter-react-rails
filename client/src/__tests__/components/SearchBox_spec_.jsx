import React from 'react';
import { Provider } from 'react-redux';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createMockStore } from 'redux-test-utils';

import SearchBox, { Unwrapped as UnwrappedSearchBox } from '../../components/SearchBox';
import store from '../../redux/store';
import { changeSearchTerm } from '../../redux/actionCreators';

configure({ adapter: new Adapter() });

test('SearchBox renders correctly', () => {
  const component = shallow(<UnwrappedSearchBox searchTerm="" />);
  expect(component).toMatchSnapshot();
});

test('Calling changeSearchTerm changes the input value in the search box', () => {
  const searchWord = 'Doe';
  store.dispatch(changeSearchTerm(searchWord));
  const component = mount(<Provider store={store}>
    <SearchBox />
  </Provider>);
  const input = component.find('input');
  expect(searchWord).toEqual(input.props().value);
});

test('Changing the search box value calls changeSearchTerm', () => {
  const searchWord = 'Doe';
  const testState = {
    searchTerm: '',
  };
  const mockStore = createMockStore(testState);
  const component = mount(<Provider store={mockStore}>
    <SearchBox />
  </Provider>);
  const input = component.find('input');
  const action = {
    type: 'CHANGE_SEARCH_TERM',
    searchTerm: searchWord,
  };
  input.simulate('change', { target: { value: searchWord } });
  expect(mockStore.isActionDispatched(action)).toBe(true);
});
