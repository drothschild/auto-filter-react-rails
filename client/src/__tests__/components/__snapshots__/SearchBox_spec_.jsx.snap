// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox
    handleSearchTermChange={[Function]}
    searchTerm=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
/>,
      "className": "search-box",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": [Function],
        "placeholder": "Search",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
/>,
        "className": "search-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
